plaidml/core/ffi.cc

```diff
@@ -107,6 +108,28 @@ Type convertFromDataType(plaidml_datatype dtype, MLIRContext* context) {
   llvm_unreachable("Invalid plaidml_datatype");
 }
 
+plaidml_strings* toFFI(std::vector<std::string> strings) {
+  std::unique_ptr<plaidml_string* []> elts { new plaidml_string*[strings.size()] };
+  auto result = std::make_unique<plaidml_strings>();
+
+  std::size_t idx = 0;
+  try {
+    while (idx < strings.size()) {
```

Comment:
Can we just use a for loop?

Reply:
Probably.  What's the goal?

The important bit is to keep track of where we are in `elts`, for the unwind.  I wrote this the first way that came to mind here and moved on to the next bit of coding, but there're probably ways to improve it.

Looking at it, I'm pretty sure the compiler can optimize it just fine (except that we should probably explicitly capture `strings.size()` if we really care), and I kinda like how clear it is about its use of `strings[idx]` as an rvalue in the move; I think using a ranged for loop over `strings` would work, but I don't know that it's as obviously correct, since then the cursor is working more as an lvalue except where we'd be putting it through `std::move()`.

A better approach might be to refactor `plaidml_strings` so that it works a bit more like a standard container, with proper RAII semantics, while preserving its FFI utility.  We might have bigger fish to fry, though...

Reply:
Code clarity, not a big deal.